src/sfuncs/dist/cat.jl

```diff
@@ -1,7 +1,33 @@
-export Cat
+export 
+    Cat,
+    Categorical,
+    Discrete
 
 import ..Utils.normalize
-import Distributions.Categorical
+import Distributions
+
+const Categorical{P, Ps} = DistributionsSF{Distributions.Categorical{P, Ps}, Int}
+Categorical(p) = Categorical{typeof(p[1]), typeof(p)}(p)
```

Comment:
Assuming non-empty arrays/tuples is bad.  Catch this case and throw an explicit error.